classes/author/Collector.inc.php

```diff
@@ -0,0 +1,232 @@
+<?php
+/**
+ * @file classes/author/Collector.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class author
+ *
+ * @brief A helper class to configure a Query Builder to get a collection of announcements
+ */
+
+namespace PKP\author;
+
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Facades\DB;
+use PKP\core\interfaces\CollectorInterface;
+use PKP\plugins\HookRegistry;
+
+class Collector implements CollectorInterface
+{
+    public const ORDERBY_SEQUENCE = 'sequence';
+    public const ORDERBY_ID = 'id';
+
+    /** @var string The default orderBy value for authors collector */
+    public $orderBy = self::ORDERBY_SEQUENCE;
+
+    /** @var DAO */
+    public $dao;
+
+    /** @var array|null */
+    public $contextIds = null;
+
+    /** @var array|null */
+    public $publicationIds = null;
+
+    /** @var string get authors with a family name */
+    protected $familyName = null;
```

Comment:
Could you add typehints for these? e.g.: `protected ?string $familyName = null`
(The self-doc is missing the nullable aspect of the variable -- but better to use explicit typehints and remove the redundant part of the self-doc)

Reply:
Fixed. Aren't `string`s always "nullable" though?

Reply:
No, if you want a `null` you'll need to declare it `?string`:
```
<?php

class c {
    public string $s = null;
}
```
...results in...
```
Fatal error: Default value for property of type string may not be null. Use the nullable type ?string to allow null default value in .../test.php on line 4
```
This is with PHP 8.0.2; earlier versions of PHP might not fail/warn.